collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,382 @@
+
+And on that singleton there are now handy methods to get a direct reference to 2D and 3D viewports, thanks to a contribution from [Cory Petkovsek](https://github.com/TokisanGames) ([GH-68696](https://github.com/godotengine/godot/pull/68696)).
+
+Finally, editor plugins can now trigger the warning dialog, prompting the user about unsaved changes when closing the editor ([GH-67503](https://github.com/godotengine/godot/pull/67503) by [Tomasz Chabora](https://github.com/KoBeWi)).
+
+
+### GDExtension
+
+One of the more powerful Godot features is the GDExtension system, providing a framework for low-level engine extensions which don't require being bundled with the engine. This serves multiple purposes, from basic extended functionality and integration with third-party libraries, to introduction of additional scripting languages or physics backends. Basically extensions are supposed to do everything that engine modules can, but can be compiled and distributed independently.
+
+As such, with every release one of the goals of the GDExtension team is to work on better parity between extensions and modules, and Godot 4.2 is not an exception. Notable improvements include unexposed class registration by [Daylily-Zeleen](https://github.com/Daylily-Zeleen) ([GH-70329](https://github.com/godotengine/godot/pull/70329)), custom `Callable` support by [Mai Lavelle](https://github.com/maiself) ([GH-79005](https://github.com/godotengine/godot/pull/79005)), indexed properties by [Mikael Hermansson](https://github.com/mihe) ([GH-79763](https://github.com/godotengine/godot/pull/79763)), and advanced registered property management with `validate_property()` by [David Snopek](https://github.com/dsnopek) ([GH-81261](https://github.com/godotengine/godot/pull/81261), [GH-81515](https://github.com/godotengine/godot/pull/81515)).
+
+This release also contains a major and long-awaited improvement to the development workflow of extensions. After some initial work by [George Marques](https://github.com/vnen) ([GH-80188](https://github.com/godotengine/godot/pull/80188)), [David Snopek](https://github.com/dsnopek) was able to implement in-editor hot reloading ([GH-80284](https://github.com/godotengine/godot/pull/80284))! Where previously an editor restart was required to pick up on the changes in a GDExtension library, updates can now be handled on the fly. This change also makes C++ scripting with extensions a more viable tool.
+
+In other big news, dynamic libraries can now be loaded on the Web platform, which means GDExtensions can be used with web exports ([GH-82633](https://github.com/godotengine/godot/pull/82633)). Thanks to [Fabio Alessandrelli](https://github.com/Faless) for his outstanding work!
```

Comment:
There are still a few quirks, but there's good work being made on debugging and solving them, so it might be fine not to mention them. This statement should become fully accurate in future betas.

Reply:
I would rather mention it and add your first sentence as an extra clause :)